atom/browser/api/atom_api_web_contents.cc

```diff
@@ -919,10 +923,6 @@ bool WebContents::OnMessageReceived(const IPC::Message& message) {
 // be destroyed on close, and WebContentsDestroyed would be called for it, so
 // we need to make sure the api::WebContents is also deleted.
 void WebContents::WebContentsDestroyed() {
-  // This event is only for internal use, which is emitted when WebContents is
-  // being destroyed.
-  Emit("will-destroy");
```

Comment:
> however for normal window now it is emitted after WebContents gets destroyed.

I just tested for normal window close and saw the event emitted before `WebContentsDestroyed` is called. The change here is that either scenario 1 or 3 of webContents destruction it will always invoke `will-destroy` before anything else and follow that `InspectebWebContentsImpl` gets to do the cleanup first. Is there something else that I missed ?